library/src/scala/annotation/transparentTrait.scala

```diff
@@ -0,0 +1,8 @@
+package scala.annotation
+
+/** An annotation that can be used from Scala 2 to mark a trait as transparent.
+ *  Scala 3 code would use the modifier `transparent` instead. Transparent traits
+ *  are not inferred when combined with other types in an intersection.
+ *  See reference/other-new-features/transparent-traits.html for details.
+ */
+final class transparentTrait extends StaticAnnotation
```

Comment:
What are the use cases for this feature?

Reply:
Essentially cross-compilation. stdLib213 already contains 9 traits that are annotated with `@transparentTrait`. Right now, we treat them also specially in a "assume transparent" set. But once we can backport the code into the standard library, we can drop that special treatment.

Reply:
The issue seems to be that we won't be able to drop the annotation itself. I'm not sure if that is a good idea.